lib/block.rb

```diff
@@ -0,0 +1,15 @@
+require 'date'
+
+module HotelBooking
+  class Block
+    attr_reader :check_in, :check_out, :rooms, :discounted_rate
+
+    def initialize(check_in, check_out, rooms = [], discounted_rate)     
+      @check_in = check_in
```

Comment:
When a class only has instance variables and no methods, either it doesn't need to be a class, or you're doing that class's work somewhere else. This class should be able to make its own reservations, and report information about when it is available.